src/App.js

```diff
@@ -49,16 +51,26 @@ export default class App extends React.Component {
 		this.setInStorage('theme', theme);
 	};
 	addToPhotos = (photo) => {
-		let savedPhotos = [...this.state.savedPhotos, photo];
-		this.setState({ savedPhotos: savedPhotos });
-		this.setInStorage('savedPhotos', savedPhotos);
+		if (
+			this.state.savedPhotos.filter((saved) => saved.id === photo.id).length ===
+			0
+		) {
+			let savedPhotos = [...this.state.savedPhotos, photo];
+			this.setState({ savedPhotos: savedPhotos });
+			this.setInStorage('savedPhotos', savedPhotos);
+		}
```

Comment:
Check if the photo already exists in savedPhotos

Reply:
you can use .find instead of filter